store/contracts.ts

```diff
@@ -1,15 +1,217 @@
-import { Module, VuexModule } from "vuex-module-decorators"
+import { Action, Module, Mutation, VuexModule } from "vuex-module-decorators"
+import { ethers } from "ethers"
+import LSPAbi from "~/abis/LSP.json"
+import { getCurrentProvider } from "~/store/web3"
 import { LSPConfiguration } from "~/types"
 
+const abis: Record<string, Array<string>> = require("~/abis")
+const addresses: Record<string, string> = require("~/addresses.json")
+
+type SyntheticTokenContractMapping = {
+  shortContract: ethers.Contract
+  longContract: ethers.Contract
+}
+
+type SyntheticTokenBalances = {
+  shortBalance: number
+  longBalance: number
+}
+
+const lspContracts: Record<string, ethers.Contract> = {}
+const collateralContracts: Record<string, ethers.Contract> = {}
+const syntheticTokenContracts: Record<string, SyntheticTokenContractMapping> =
+  {}
+
 @Module({
   stateFactory: true,
   name: "contracts",
   namespaced: true,
 })
 export default class contracts extends VuexModule {
   contractConfigs: Array<LSPConfiguration> = require("~/deployedContractConfigs.json")
+  syntheticTokenBalances: Record<string, SyntheticTokenBalances> = {}
+  collateralTokenBalances: Record<string, number> = {}
+  contractsInitialized: boolean = false
+  tokenBalancesLoaded: boolean = false
 
   get syntheticNames() {
     return this.contractConfigs.map((config) => config.syntheticName)
   }
+
+  get getCollateralTokenBalances(): Record<string, number> {
+    return this.collateralTokenBalances
+  }
+
+  get getSyntheticTokenBalances(): Record<string, SyntheticTokenBalances> {
+    return this.syntheticTokenBalances
+  }
+
+  @Mutation
+  setTokenBalancesLoaded(loaded: boolean) {
+    this.tokenBalancesLoaded = loaded
+  }
+
+  @Mutation
+  setCollateralTokenBalance(payload: {
+    collateralName: string
+    collateralBalance: number
+  }) {
+    const { collateralName, collateralBalance } = payload
+    console.log(
+      `Set collateral balance of ${collateralName} to ${collateralBalance}`
```

Comment:
I know it is programming 101 to use console.log for sanity checks, but we should standardize our debugging messages after mvp 1.0 release

Reply:
Agreed.